tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -970,6 +971,10 @@ create_compressed_scan_paths(PlannerInfo *root, RelOptInfo *compressed_rel, int
 {
 	Path *compressed_path;
 
+	/* clamp total_table_pages to 10 pages since this is the
+	 * minimum estimate for number of pages */
+	root->total_table_pages = Max(compressed_rel->pages, 10);
```

Comment:
why is this not ``` root->total_table_pages = compressed_rel->pages ```. If we don't have the compressed_rel size yet, shouldn't that be estimated before the assignment here?

Reply:
estimate_rel_size will have been called at that point for the rel